src/Admin/AbstractAdmin.php

```diff
@@ -2005,16 +2005,28 @@ public function isGranted($name, ?object $object = null): bool
 
     public function getUrlSafeIdentifier(object $model): string
     {
-        return $this->getModelManager()->getUrlSafeIdentifier($model);
+        try {
+            return $this->getModelManager()->getUrlSafeIdentifier($model);
+        } catch (\InvalidArgumentException $e) {
+            return '';
+        }
     }
 
     public function getNormalizedIdentifier(object $model): string
     {
-        return $this->getModelManager()->getNormalizedIdentifier($model);
+        try {
+            return $this->getModelManager()->getNormalizedIdentifier($model);
+        } catch (\InvalidArgumentException $e) {
+            return '';
+        }
     }
 
-    public function id(object $model): string
+    public function id(?object $model): string
```

Comment:
This shouldn't be called with `null`.

What's your use case ? There was none in the issue #6535

Reply:
> This shouldn't be called with `null`.
> 
> What's your use case ? There was none in the issue #6535

Because if you look in the template coming from [doctrine-orm-admin-bundle/Resources/views/Form/form_admin_fields.html.twig](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/5e602ba1871431ec3cc19e7421e674c5471b7e90/src/Resources/views/Form/form_admin_fields.html.twig#L59) on line 59 this function is called with `null` when you create new record. This can be fixed very easily if we make check in the template before calling the function.
Something like this:
`{% if sonata_admin.value and sonata_admin.field_description.associationadmin.id(sonata_admin.value) %}`
the same need to be done for line 62 where is the `objectId`.

I am quite new to Sonata and I don't know if you have other use cases like this one coming from other bundles and that's why I suggest to allow `null`. If you don`t have cases like this one we can fix the model list template. I can make PR to SonataDoctrineORMAdminBundle for this if you agree.

For the other things I will follow your advice and I will make new PRs.

Reply:
IMHO 
```
{% if sonata_admin.value and sonata_admin.field_description.associationadmin.id(sonata_admin.value) %}
```
Is the right thing to do.

There is no reason to call an `id` method on a non-object. We shouldn't allow `null` just because it was (badly) used before. We're trying to improve our Interface, so it's better to fix the `id` calls.

I made https://github.com/sonata-project/SonataAdminBundle/pull/6578 on SonataAdmin to fix this on 3.x.
Can you do the PR on DoctrineORMAdmin ?
(The check is not needed line 62 because the code inside the if won't be executed with the check you'll add)